src/controls/gamepad-controls.js

```diff
@@ -273,12 +273,20 @@ module.exports = AFRAME.registerComponent('gamepad-controls', {
   getJoystick: function (index, target) {
     const gamepad = this.getGamepad();
     if (gamepad.mapping === 'xr-standard') {
+      const handedness = this.system.controllers[this.data.controller].handedness;
+      if (handedness === 'right') {
+        this.data.controller = 1;
+      }
```

Comment:
Could this block of code be moved above the `const gamepad = this.getGamepad();` line? In that case it would be applied in time to affect the `getGamepad()` call, and only needs to be written once. 

I think it's also best to do `setAttribute('controller', 1)` rather than manipulating `this.data` directly.

Reply:
That's an approach I tried initially so that I could use a single conditional statement for both the `handedness` and the `hand` bindings (and, consequently, both an `xrController` or a `stdGamepad` could be detected).

By doing so, though, the code throws an error.

In, particular, not only I get an error if I move the code **above** the `const gamepad = this.getGamepad();` line

(note: the code below is from a modded `aframe-extras` I am using for testing locally)

![above-get-gamepad](https://user-images.githubusercontent.com/30973948/89402579-01a08600-d70f-11ea-8518-1786fb748b2f.png)

but the same error also shows up if I move the code block **below** that line

![below-get-gamepad](https://user-images.githubusercontent.com/30973948/89402646-1715b000-d70f-11ea-868d-464b804da674.png)

Therefore the only way to get it to work correctly, apparently, is to move the 2 bindings `handedness` and `hand` and their respective conditional statements inside the `if (gamepad.mapping === 'xr-standard') { }` conditional statement.

Below you can see the updated/pushed code, which includes the other change you suggested, that is to do `setAttribute()` rather than manipulating `this.data` directly.

![gamepad-controls-working-code-updated](https://user-images.githubusercontent.com/30973948/89402868-64921d00-d70f-11ea-8f8d-1489e7529ef2.png)

LMK if there are other alternative approaches you would like me to try, Don, or if you are happy with the updated/pushed code and still willing to merge this PR.

Thanks!

Reply:
Would this version work, before the if/else blocks?

```js
const controller = this.system.controllers[this.data.controller];
if (controller && controller.handedness === 'right') {
  this.el.setAttribute('gamepad-controls', 'controller', 1);
}
const gamepad = this.getGamepad();
// ...
```

Reply:
Yes, it does. Thanks for your guidance!

Following the same approach you suggested for the `xrController`, I tried to move the `stdGamepad` binding as well to use a single conditional statement and remove the same 3 lines from the `else` block below.

Unfortunately, this version works only on Rift, whereas it throws a `Cannot read property 'hand' of null` error on Quest:

```JavaScript
const xrController = this.system.controllers[this.data.controller],
      stdGamepad = navigator.getGamepads;
if (xrController && xrController.handedness === 'right' || 
    stdGamepad && navigator.getGamepads()[this.data.controller].hand === 'right') {
  this.el.setAttribute('gamepad-controls', 'controller', 1);
}
const gamepad = this.getGamepad();
//...
```

Therefore the final code, tested/working both on Quest and on Rift (Firefox without/with `dom.vr.webxr.enabled` enabled), would be:

```JavaScript
const xrController = this.system.controllers[this.data.controller];
if (xrController && xrController.handedness === 'right') {
  this.el.setAttribute('gamepad-controls', 'controller', 1);
}
const gamepad = this.getGamepad();

if (gamepad.mapping === 'xr-standard') {
  // See: https://github.com/donmccurdy/aframe-extras/issues/307
  switch (index) {
    case Joystick.MOVEMENT: return target.set(gamepad.axes[2], gamepad.axes[3]);
    case Joystick.ROTATION: return target.set(gamepad.axes[0], gamepad.axes[1]);
  }
} else {
  const stdGamepad = navigator.getGamepads && navigator.getGamepads()[this.data.controller];
  if (stdGamepad.hand === 'right') {
    this.el.setAttribute('gamepad-controls', 'controller', 1);
  }
  switch (index) {
    case Joystick.MOVEMENT: return target.set(gamepad.axes[0], gamepad.axes[1]);
    case Joystick.ROTATION: return target.set(gamepad.axes[2], gamepad.axes[3]);
  }
}
//...
```

LMK if there is anything I might have overlooked or if you are happy with this last version I just committed and pushed.

Thanks again, Don. 🙂